[ADDED] Activity type and location source found in real world data.

hossain-khan opened the pull request:
# Summary
Added some types that were observed in Google Location History Takeout data dump.

hossain-khan committed:
[ADDED] Activity type and location source found in real world data.

hossain-khan commented on schemas/Records.schema.json:
```diff
@@ -333,6 +333,10 @@
                         "cell",
                         "GPS",
                         "gps",
+                        VISIT_ARRIVAL,
+                        visit_arrival,
```
```json
{
  "latitudeE7": 436468347,
  "longitudeE7": -793912234,
  "accuracy": 109,
  "source": "VISIT_ARRIVAL",
  "deviceTag": -1158676519,
  "deviceDesignation": "PRIMARY",
  "timestamp": "2018-01-23T14:03:34.135Z"
}
```

hossain-khan commented on schemas/Records.schema.json:
```diff
@@ -333,6 +333,10 @@
                         "gps",
+                        VISIT_ARRIVAL,
+                        visit_arrival,
+                        VISIT_DEPARTURE,
+                        visit_departure,
```
```json
{
  "latitudeE7": 436466804,
  "longitudeE7": -793918201,
  "accuracy": 110,
  "source": "VISIT_DEPARTURE",
  "deviceTag": -1158676519,
  "deviceDesignation": "PRIMARY",
  "timestamp": "2018-01-23T21:59:35.998Z"
}
```

hossain-khan commented on schemas/Semantic.schema.json:
```diff
@@ -890,10 +890,18 @@
-                    "extra_icon": "activity_types/moving_24dp.svg"
+                    "extra_icon": "activity_types/unknown.svg"
+                },
+                {
+                    "type": "string",
+                    "const": "UNKNOWN",
+                    "title": "Uknown Activity",
+                    "description": "Type used when unable to detect the current activity.",
+                    "extra_color": "#03a9f4",
+                    "extra_icon": "activity_types/unknown.svg"
                 },
```
```json
{
  "type": "UNKNOWN",
  "confidence": 23
}
```

https://developers.google.com/android/reference/com/google/android/gms/location/DetectedActivity#UNKNOWN

hossain-khan commented on schemas/Semantic.schema.json:
```diff
@@ -911,6 +919,14 @@
                     "extra_color": "#03a9f4",
                     "extra_icon": "activity_types/directions_walk_24dp.svg"
                 },
+                {
+                    "type": "string",
+                    "const": "ON_FOOT",
+                    "title": "On Foot",
+                    "description": "",
+                    "extra_color": "#4db6ac",
+                    "extra_icon": "activity_types/directions_walk_24dp.svg"
+                },
```
```json
{
  "type": "ON_FOOT",
  "confidence": 1
}
```

hossain-khan commented on schemas/Semantic.schema.json:
```diff
@@ -919,6 +935,14 @@
                     "extra_color": "#4db6ac",
                     "extra_icon": "activity_types/directions_bike_24dp.svg"
                 },
+                {
+                    "type": "string",
+                    "const": "ON_BICYCLE",
+                    "title": "On bicycle",
+                    "description": "",
+                    "extra_color": "#4db6ac",
+                    "extra_icon": "activity_types/directions_bike_24dp.svg"
+                },
```
```json
{
  "type": "ON_BICYCLE",
  "confidence": 5
}
```

hossain-khan commented on schemas/Semantic.schema.json:
```diff
@@ -1190,6 +1214,14 @@
                     "extra_icon": "activity_types/paragliding_24dp.svg"
+                },
+                {
+                    "type": "string",
+                    "const": "TILTING",
+                    "title": "Tilting",
+                    "description": "The device angle relative to gravity changed significantly. This often occurs when a device is picked up from a desk or a user who is sitting stands up.",
+                    "extra_color": "#4db6ac",
+                    "extra_icon": "activity_types/moving_24dp.svg"
                 }
```
```json
{
  "type": "TILTING",
  "confidence": 100
}
```

hossain-khan commented on schemas/Semantic.schema.json:
```diff
@@ -1232,6 +1264,14 @@
                     "extra_color": "#03a9f4",
                     "extra_icon": "semantic_types/label_circled_24dp.svg"
+                },
+                {
+                    "type": "string",
+                    "const": "TYPE_UNKNOWN",
+                    "title": "Type Unknown Location",
+                    "description": "Location type is unknown",
+                    "extra_color": "#03a9f4",
+                    "extra_icon": "semantic_types/label_circled_24dp.svg"
                 }
```
```json
{
  "latitudeE7": 427012790,
  "longitudeE7": -788526840,
  "placeId": "Chklm2Fe-r8TK4gRkgyI7imTse0",
  "address": "Brampton, ON L7A, Canada",
  "name": "Creditview Rd s/of Kilkarrin Rd",
  "semanticType": "TYPE_UNKNOWN",
  "locationConfidence": 0.100103855,
  "calibratedProbability": 0.100103855
}
```

hossain-khan reviewed (commented):
Added snapshot of data that contains the new types.

CarlosBergillos commented:
Hi! Thanks for using the project and contributing!

I think that the `ON_BICYCLE`, `ON_FOOT`, `UNKNOWN` and `TILTING` activity types that you are adding are activity types found in the Records.json file (and already present in that schema [here](https://github.com/CarlosBergillos/LocationHistoryFormat/blob/54b0a706e44603a56df6418fdd8b944968588fd3/schemas/Records.schema.json#L451)), but they are not activity types found in the semantic files so they should not be added there.

Can you check that?

hossain-khan committed:
[FIXED] Removed wrongly added Activity Type that belongs to `Records`

Addresses https://github.com/CarlosBergillos/LocationHistoryFormat/pull/4#issuecomment-2008275584

hossain-khan commented:
> Hi! Thanks for using the project and contributing!
> 
> I think that the `ON_BICYCLE`, `ON_FOOT`, `UNKNOWN` and `TILTING` activity types that you are adding are activity types found in the Records.json file (and already present in that schema [here](https://github.com/CarlosBergillos/LocationHistoryFormat/blob/54b0a706e44603a56df6418fdd8b944968588fd3/schemas/Records.schema.json#L451)), but they are not activity types found in the semantic files so they should not be added there.
> 
> Can you check that?

Yes, you are right. I double-checked, and I have mixed up records activity with semantic activities. 
I have removed those enums. Thanks for checking.

hossain-khan committed:
[MINOR] Use same naming convention for the icons

hossain-khan commented on schemas/Semantic.schema.json:
```diff
@@ -890,10 +890,10 @@
+                    "description": "Type used when unable to detect the current activity.",
                     "extra_color": "#03a9f4",
-                    "extra_icon": "activity_types/moving_24dp.svg"
+                    "extra_icon": "activity_types/unknown_24dp.svg"
```
I didn't realize that the icons are from https://github.com/CarlosBergillos/LocationHistoryFormat/tree/main/docs/static/icons/activity_types and used in the generated docs

![Screenshot 2024-03-20 231425](https://github.com/CarlosBergillos/LocationHistoryFormat/assets/99822/31dab1f2-8a28-4559-83c5-81940feba8f5)

CarlosBergillos replied:
The icons and titles (and colors) are actually based on the Google Maps Timeline web interface:

![Screenshot_20240401_193709](https://github.com/CarlosBergillos/LocationHistoryFormat/assets/1731534/72a75c6e-008b-43b5-97de-6125f548290d)

So maybe we should keep it aligned with them? in this case "Moving".

hossain-khan commented on schemas/Semantic.schema.json:
```diff
@@ -1232,6 +1232,14 @@
+                    "title": "Type Unknown Location",
+                    "description": "Location type is unknown",
+                    "extra_color": "#03a9f4",
+                    "extra_icon": "semantic_types/unknown_24dp.svg"
```
Same here - the icon does not exists in https://github.com/CarlosBergillos/LocationHistoryFormat/tree/main/docs/static/icons/semantic_types - what should I use here?

CarlosBergillos replied:
Oh, I haven't encountered this one!
In this case (semantic type) the icons are also based on what is shown in the web interface.

![Screenshot_20240401_194925](https://github.com/CarlosBergillos/LocationHistoryFormat/assets/1731534/9d483d69-2516-4635-bdb5-ba66f4505ea2)

If you have access to the Google account where this `TYPE_UNKNOWN` was found, can you navigate to that day in the Google Maps Timeline web interface and see how it looks there?

hossain-khan replied:
It looks very similar.

![Screenshot 2024-04-13 232530](https://github.com/CarlosBergillos/LocationHistoryFormat/assets/99822/e77c7fcf-0382-4f44-af59-ebcd73d2421d)

This type shows up in the `"otherCandidateLocations"` location items. Eg.

```json
{
    "placeVisit": {
        "location": {
            "semanticType": "TYPE_HOME",
            "locationConfidence": 100.0,
            "calibratedProbability": 100.0
        },
        "duration": {
            "startTimestamp": "2024-03-04T02:34:29Z",
            "endTimestamp": "2024-03-04T13:42:14Z"
        },
        "placeConfidence": "USER_CONFIRMED",
        "visitConfidence": 100,
        "otherCandidateLocations": [
            {
                "placeId": "Ch...wY",
                "address": "Address, City, State 12345, Country",
                "name": "Place Name",
                "semanticType": "TYPE_UNKNOWN",
                "locationConfidence": 37.953373,
                "calibratedProbability": 37.953373
            }
        ],
        "editConfirmationStatus": "CONFIRMED",
        "locationConfidence": 100,
        "placeVisitType": "SINGLE_PLACE",
        "locationAssertionType": "WITHIN_OR_AT",
        "lastEditedTimestamp": "2024-03-13T00:37:59.560Z",
        "placeVisitImportance": "MAIN"
    }
}
```

hossain-khan committed:
Merge branch 'CarlosBergillos:main' into hk/update-json-schema

hossain-khan committed:
[FIXED] New enum string type error

hossain-khan committed:
[ADDED] New location source type found from another users's export

hossain-khan commented on schemas/Records.schema.json:
```diff
@@ -333,6 +333,12 @@
                         "cell",
                         "GPS",
                         "gps",
+                        "MANUAL",
+                        "manual",
```
@CarlosBergillos  Found another new type from different user's export.

```json
{
    "latitudeE7": 429362606,
    "longitudeE7": -778308950,
    "accuracy": 4,
    "velocity": 0,
    "altitude": 169,
    "verticalAccuracy": 3,
    "source": "MANUAL",
    "deviceTag": -379135184,
    "platformType": "IOS",
    "serverTimestamp": "2023-11-30T22:30:02.016Z",
    "deviceTimestamp": "2023-11-30T22:30:01.943Z",
    "batteryCharging": false,
    "formFactor": "PHONE",
    "timestamp": "2023-11-30T22:30:01Z"
}
```

91680 commented:
> Hi! Thanks for using the project and contributing!
> 
> I think that the `ON_BICYCLE`, `ON_FOOT`, `UNKNOWN` and `TILTING` activity types that you are adding are activity types found in the Records.json file (and already present in that schema [here](https://github.com/CarlosBergillos/LocationHistoryFormat/blob/54b0a706e44603a56df6418fdd8b944968588fd3/schemas/Records.schema.json#L451)), but they are not activity types found in the semantic files so they should not be added there.
> 
> Can you check that?
